src/agentic/actor_agents.py

```diff
@@ -1266,51 +1223,115 @@ def ensure_api_key_for_model(self, model: str):
+                        func.get_agent()._agent, 
+                        handoff=True
+                    )
+                )
+            elif isinstance(func, BaseAgentProxy):
+                useable.append(
+                    AddChild(
+                        func.name,
+                        func._agent,
+                    )
+                )
+            else:
+                tool_registry.ensure_dependencies(func)
+                useable.append(func)
+
+        return useable
+        
+    def _update_state(self, state: dict):
+        """Update the agent's state"""
+        # To be overridden by subclasses
+        pass
+
+    def _set_agent_debug_level(self, debug_level):
+        """Set the debug level on the agent implementation"""
+        # To be overridden by subclasses
+        pass
+
+    def _reset_agent_history(self):
+        """Reset the agent's conversation history"""
+        # To be overridden by subclasses
+        pass
+
+    def _get_agent_history(self):
+        """Get the agent's conversation history"""
+        # To be overridden by subclasses
+        pass
+
+    def _create_agent_instance(self, request_id: str):
+        """Create a new agent instance for a request"""
+        # This is implemented by the subclasses (e.g., RayAgentProxy, LocalAgentProxy)
+        raise NotImplementedError("Subclasses must implement _create_agent_instance")
+
+    def _get_agent_for_request(self, request_id: str):
+        """Get the agent instance for a request, creating it if needed"""
+        if request_id not in self.agent_instances:
+            self.agent_instances[request_id] = self._create_agent_instance(request_id)
+        return self.agent_instances[request_id]
+
+    def _cleanup_agent_instance(self, request_id: str):
+        """Clean up an agent instance after a request is complete"""
+        if request_id in self.agent_instances:
+            del self.agent_instances[request_id]
+
+    def start_request(self, request: str, request_context: dict = {}, 
+                     continue_result: dict = {}, run_id: Optional[str] = None,
+                     debug: DebugLevel = DebugLevel(DebugLevel.OFF)) -> StartRequestResponse:
+        """Start a new agent request"""
         self.debug.raise_level(debug)
-        self.queue_done_sentinel = "QUEUE_DONE"
 
         if not hasattr(depthLocal, 'depth'):
             depthLocal.depth = 0
         else:
             depthLocal.depth += 1
 
+        # Create request ID if not provided in continue_result
+        request_id = continue_result.get("request_id") or str(uuid.uuid4())
+
         # Initialize new request
         request_obj = Prompt(
             self.name,
             request,
             debug=self.debug,
             depth=depthLocal.depth,
             request_context=request_context,
+            request_id=request_id,
         )
 
-        # Re-initialize run tracking
-        self.init_run_tracking(self.db_path, run_id)
```

Comment:
Run tracking init is moved into the `next_turn` method which itself instantiates the agent object specific to the request id. we need that to attach the callback to the underlying agent.